tests/modules/omics/test_omics_workflow_fetcher.py

```diff
@@ -0,0 +1,115 @@
+from typing import Callable
+import pytest
+from dataall.base.db import Engine
+from dataall.modules.omics.tasks.omics_workflows_fetcher import fetch_omics_workflows
+from dataall.modules.omics.db.omics_repository import OmicsRepository
+
+
+@pytest.fixture
+def second_environment(env, org_fixture, group):
+    yield env(
+        org=org_fixture,
+        account='222222222222',
+        envname='second_environment',
+        owner=group.owner,
+        group=group.name,
+        role='new-role',
+    )
+
+
+def test_omics_workflow_fetcher_new_workflows_single_environment(db: Engine, module_mocker, env_fixture):
+    """Checks that new workflows are added to the RDS database"""
+
+    # Given one environment and 2 READY2RUN workflows returned from that account
+    items = [
+        {'arn': 'some-arn-1', 'id': 'id-1', 'name': 'name-1', 'status': 'ACTIVE', 'type': 'READY2RUN'},
+        {'arn': 'some-arn-2', 'id': 'id-2', 'name': 'name-2', 'status': 'ACTIVE', 'type': 'READY2RUN'},
+    ]
+    module_mocker.patch(
+        'dataall.modules.omics.aws.omics_client.OmicsClient.client',
+        return_value=True,
+    )
+    mocker = module_mocker.patch(
+        'dataall.modules.omics.aws.omics_client.OmicsClient.list_workflows',
+        return_value=items,
+    )
+    # When we run the omics workflows fetcher
+    success = fetch_omics_workflows(db)
+    # Then, the task completes successfully
+    assert success == True
+    # Then, the mocker is called only once
+    mocker.assert_called_once()
+    with db.scoped_session() as session:
+        workflows = OmicsRepository(session).paginated_omics_workflows(filter={})
+        # Then, the 2 workflows are added to RDS
+        assert workflows.get('count') == 2
+        # Finally, clean_up test
+        for workflow in workflows.get('nodes'):
+            session.delete(workflow)
+
+
+def test_omics_workflow_fetcher_new_workflows_multiple_environments(
+    db: Engine, module_mocker, env_fixture, second_environment
+):
+    """Checks that new workflows are added to the RDS database WITHOUT duplicating the workflows of both environments"""
+
+    # Given 2 environment and 2 READY2RUN workflows returned from each of the accounts
+    items = [
+        {'arn': 'some-arn-1', 'id': 'id-1', 'name': 'name-1', 'status': 'ACTIVE', 'type': 'READY2RUN'},
+        {'arn': 'some-arn-2', 'id': 'id-2', 'name': 'name-2', 'status': 'ACTIVE', 'type': 'READY2RUN'},
+    ]
+    module_mocker.patch(
+        'dataall.modules.omics.aws.omics_client.OmicsClient.client',
+        return_value=True,
+    )
+    mocker = module_mocker.patch(
+        'dataall.modules.omics.aws.omics_client.OmicsClient.list_workflows',
+        return_value=items,
+    )
+    # When we run the omics workflows fetcher
+    success = fetch_omics_workflows(db)
+    # Then, the task completes successfully
+    assert success == True
+    # Then, the mocker is called twice
+    assert mocker.call_count == 2
```

Comment:
with workflow fetcher changes the call count will still be 1 - can change to `mocker.assert_called_once()`

Reply:
@dlpzx would you mind taking this one? I didn't want to change and break your test

Reply:
I will do 1 last review - but I think will be good to approve this PR after this 1 final change

Reply:
All done. I also added some guardrails to clean-up each test. In the config.json I enabled the module so that any PR executes the tests, but I want to know your opinion